src/react-components/profile-entry-panel.js

```diff
@@ -84,26 +76,31 @@ class ProfileEntryPanel extends Component {
   };
 
   setAvatarFromMediaResult = ({ detail: entry }) => {
-    this.setState({
-      avatarId: entry.id,
-      avatarType: getAvatarType(entry.id),
-      avatarGltfUrl: entry.gltfs.avatar
-    });
+    this.setState({ avatarId: entry.id });
     // Replace history state with the current avatar id since this component gets destroyed when we open the
     // avatar editor and we want the back button to work. We read the history state back via the avatarId prop.
     // We read the current state key from history since it could be "overlay" or "entry_step".
     replaceHistoryState(this.props.history, this.props.history.location.state.key, "profile", { avatarId: entry.id });
   };
 
-  async componentDidMount() {
-    this.setState({ avatarGltfUrl: await getAvatarGltfUrl(this.state.avatarId) });
+  componentDidMount() {
     if (this.nameInput) {
       // stop propagation so that avatar doesn't move when wasd'ing during text input.
       this.nameInput.addEventListener("keydown", this.stopPropagation);
       this.nameInput.addEventListener("keypress", this.stopPropagation);
       this.nameInput.addEventListener("keyup", this.stopPropagation);
     }
     this.scene.addEventListener("action_selected_media_result_entry", this.setAvatarFromMediaResult);
+    // This handles editing avatars in the entry_step, since this component remains mounted with the same avatarId
```

Comment:
This comment is a bit misleading, since the profile panel _does_ get unmounted when you edit an avatar, but surprisingly, the action_avatar_saved event is emitted _after_ the profile panel is re-mounted, so this handler ends up working anyway.

Reply:
It actually does not get unmounted in the entry flow (when you haven't set your name) case